src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -1348,7 +1386,27 @@ private void AssignUnscheduledRequestToNode(SchedulableRequest request, int node
 
             responses.Add(ScheduleResponse.CreateScheduleResponse(nodeId, request.BuildRequest, mustSendConfigurationToNode));
             TraceScheduler("Executing request {0} on node {1} with parent {2}", request.BuildRequest.GlobalRequestId, nodeId, (request.Parent == null) ? -1 : request.Parent.BuildRequest.GlobalRequestId);
+
+            WarnWhenProxyBuildsGetScheduledOnOutOfProcNode();
+
             request.ResumeExecution(nodeId);
+
+            void WarnWhenProxyBuildsGetScheduledOnOutOfProcNode()
+            {
+                if (request.BuildRequest.ProxyTargets != null && nodeId != InProcNodeId)
+                {
+                    ErrorUtilities.VerifyThrow(
+                        _componentHost.BuildParameters.DisableInProcNode || _forceAffinityOutOfProc,
+                        "Proxy requests should only get scheduled to out of proc nodes when the inproc node is disabled");
+
+                    var loggedWarnings = Interlocked.CompareExchange(ref _loggedWarningsForProxyBuildsOnOutOfProcNodes, 1, 0);
```

Comment:
If this is happening only in the scheduler node, shouldn't this always fire once even without this?

Reply:
Oh, so because of intra-process parallelism, this one node can fire several warnings. Makes sense, thanks!

Reply:
Yup, the main node is the only one hosting a scheduler, and whenever the scheduler would schedule a proxy build it would trigger this warning, if we wouldn't inhibit it.